lighty-core/lighty-controller/src/main/java/io/lighty/core/controller/impl/util/InitialDataImportUtil.java

```diff
@@ -0,0 +1,104 @@
+                            rootSchemaNode.getQName()));
+            try (NormalizedNodeStreamWriter writer = ImmutableNormalizedNodeStreamWriter.from(builder)) {
+                try (JsonParserStream jsonParser =
+                             JsonParserStream.create(writer,
+                                     JSONCodecFactorySupplier.DRAFT_LHOTKA_NETMOD_YANG_JSON_02
+                                             .getShared(services.getDOMSchemaService().getGlobalContext()))) {
+                    try (JsonReader reader =
+                                 new JsonReader(new InputStreamReader(dataStream, Charset.defaultCharset()))) {
+                        jsonParser.parse(reader);
+                        NormalizedNode<?, ?> nodes = builder.build();
+                        DOMDataTreeWriteTransaction wrTrx = services.getClusteredDOMDataBroker()
+                                .newWriteOnlyTransaction();
+                        wrTrx.merge(LogicalDatastoreType.CONFIGURATION, YangInstanceIdentifier.empty(), nodes);
+                        wrTrx.commit().get(IMPORT_TIMEOUT_MILLIS, TimeUnit.MILLISECONDS);
+                        LOG.debug("Normalized nodes loaded on startup from file {} :\n {}",
+                                initialConfigDataFile.getAbsolutePath(), nodes);
+                    }
+                }
+            }
+        }
+    }
+
+    public static void importInitialConfigDataFile(@NonNull File file, @NonNull LightyServices services)
+            throws InterruptedException, ExecutionException, TimeoutException, IOException, SerializationException,
+            IllegalStateException {
+        String extension = FilenameUtils.getExtension(file.getAbsolutePath());
+        if (extension.equalsIgnoreCase("json")) {
+            importConfigDatastoreFromJSON(file, services);
+        } else if (extension.equalsIgnoreCase("xml")) {
+            importConfigDatastoreFromXML(file, services);
+        } else {
+            throw new UnsupportedOperationException("File extension"
+                    + extension
+                    + " is not supported as initial config data format");
+        }
+    }
+
+    private static void importConfigDatastoreFromXML(File initialConfigDataFile, LightyServices services)
+            throws IOException, InterruptedException, ExecutionException, TimeoutException, SerializationException {
+        LOG.info("Loading data into config datastore from file {}", initialConfigDataFile.getAbsolutePath());
+        SchemaNode rootSchemaNode = DataSchemaContextTree.from(services.getSchemaContextProvider()
+                .getSchemaContext()).getRoot().getDataSchemaNode();
+
+        try (Reader reader =
+                     new InputStreamReader(new FileInputStream(initialConfigDataFile), Charset.defaultCharset())) {
+            NormalizedNode<?, ?> nodes = services.getXmlNodeConverter().deserialize(rootSchemaNode, reader);
```

Comment:
Create instance of XmlNodeConverter (and JsonNodeConverted) directly in these methods. Do not create it in LightyController.